components/ord-service/src/test/java/com/sap/cloud/cmp/ord/service/filter/aggregator/JsonArrayElementsAggregatorTest.java

```diff
@@ -0,0 +1,68 @@
+package com.sap.cloud.cmp.ord.service.filter.aggregator;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class JsonArrayElementsAggregatorTest {
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    @Test
+    public void testAggregate_ReturnsUnmodifiedJson_WhenArrayElementNotFound() throws Exception {
+        JsonArrayElementsAggregator aggregator = new JsonArrayElementsAggregator(mapper);
+
+        String content = "{\"@odata.context\":\"$metadata#apis\",\"value\":[{\"ordId\":\"test-id\"}]}";
+        String expectedContent  = "{\"@odata.context\":\"$metadata#apis\",\"value\":[{\"ordId\":\"test-id\"}]}";
+
+        JsonNode jsonTree = mapper.readTree(content);
+        aggregator.aggregate(jsonTree);
+        String actualContent = mapper.writeValueAsString(jsonTree);
+
+        assertEquals(expectedContent, actualContent);
+    }
+
+    @Test
+    public void testAggregate_ReturnsUnmodifiedJson_WhenArrayElementFoundButNotKnown() throws Exception {
+        JsonArrayElementsAggregator aggregator = new JsonArrayElementsAggregator(mapper);
+
+        String content = "{\"@odata.context\":\"$metadata#apis\",\"value\":[{\"unknownArray\":[{\"value\":\"automotive\"},{\"value\":\"finance\"}]}]}";
+        String expectedContent  = "{\"@odata.context\":\"$metadata#apis\",\"value\":[{\"unknownArray\":[{\"value\":\"automotive\"},{\"value\":\"finance\"}]}]}";
+
+        JsonNode jsonTree = mapper.readTree(content);
+        aggregator.aggregate(jsonTree);
+        String actualContent = mapper.writeValueAsString(jsonTree);
+
+        assertEquals(expectedContent, actualContent);
+    }
+
+    @Test
+    public void testAggregate_ReturnsModifiedJson_WhenArrayElementFound() throws Exception {
+        JsonArrayElementsAggregator aggregator = new JsonArrayElementsAggregator(mapper);
+
+        String content = "{\"@odata.context\":\"$metadata#apis\",\"value\":[{\"tags\":[{\"value\":\"automotive\"},{\"value\":\"finance\"}]}]}";
+        String expectedContent = "{\"@odata.context\":\"$metadata#apis\",\"value\":[{\"tags\":[\"automotive\",\"finance\"]}]}";
+
+        JsonNode jsonTree = mapper.readTree(content);
+        aggregator.aggregate(jsonTree);
+        String actualContent = mapper.writeValueAsString(jsonTree);
+
+        assertEquals(expectedContent, actualContent);
+    }
+
+    @Test
+    public void testAggregate_ReturnsUnmodifiedJson_WhenLabelsFound() throws Exception {
```

Comment:
Wrong test name. 

Also, I think it makes sense to have tests for nested resources for example labels and arrays in API which is part of a package.

Reply:
Yep, that's a good idea!